geth/common/types.go

```diff
@@ -119,21 +98,24 @@ type AccountManager interface {
 	ReSelectAccount() error
 
 	// SelectedAccount returns currently selected account
-	SelectedAccount() (*SelectedExtKey, error)
+	SelectedAccount() (*account.SelectedExtKey, error)
 
 	// Logout clears whisper identities
 	Logout() error
 
 	// Accounts returns handler to process account list request
 	Accounts() ([]common.Address, error)
 
-	// AccountsRPCHandler returns RPC wrapper for Accounts()
-	AccountsRPCHandler() rpc.Handler
-
 	// AddressToDecryptedAccount tries to load decrypted key for a given account.
 	// The running node, has a keystore directory which is loaded on start. Key file
 	// for a given address is expected to be in that directory prior to node start.
 	AddressToDecryptedAccount(address, password string) (accounts.Account, *keystore.Key, error)
+
+	// AccountManager returns reference to node's account manager
```

Comment:
Can't you embed `account.ServiceProvider` here?

Reply:
That is `common/types` package. As I can understand, it shouldn't have any external dependencies.
Though, I'm fairly sceptical about us needed such package at all.